### PR TITLE
openthread: New libs - increased child count, fixed NCP log

### PR DESCRIPTION
**Pull request by pszkotak**
Improvements:
- Increased child count per parent to 32
- Increased IP addresses per child to 6
- Updated README.md
- Updated NCP libraries to include the otPlatLog function
- Add libraries for NCP sample to communicate with host

This PR should replace: https://github.com/nrfconnect/sdk-nrfxlib/pull/304
It contains one more fix.

Having merged just one would help a bit in keeping the repo smaller.

Signed-off-by: Piotr Szkotak <piotr.szkotak@nordicsemi.no>

### CODE REVIEW

**Review by MarekPorwisz — CHANGES_REQUESTED** [review]
Seems you forgot to include those new libraries per platform i. spinel-ncp.a and hdlc.a

**Comment by pszkotak** [comment]
> Seems you forgot to include those new libraries per platform i. spinel-ncp.a and hdlc.a

Done

**Review by MarekPorwisz — APPROVED** [review]

File: openthread/lib/cortex-m4/soft-float/v1.1/ftd/cc310/openthread_lib_configuration.txt [hunk 1]
```diff
@@ -40,3 +46,4 @@ CONFIG_OPENTHREAD_THREAD_VERSION=1.1
 CONFIG_OPENTHREAD_THREAD_VERSION_1_1=y
 CONFIG_OPENTHREAD_UDP_FORWARD=y
 CONFIG_OPENTHREAD_XPANID=de:ad:00:be:ef:00:ca:fe
+OT_NCP_VENDOR_HOOK_SOURCE_DIR=/home/pisz/repo/ncs/nrf/samples/openthread/ncp
```
Comment (edmont) [inline openthread/lib/cortex-m4/soft-float/v1.1/ftd/cc310/openthread_lib_configuration.txt:49]: This local folder path shouldn't go upstream.

**Review by edmont — CHANGES_REQUESTED** [review]

**Review by lmaciejonczyk — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

File: openthread/lib/cortex-m4/soft-float/v1.1/ftd/cc310/openthread_lib_configuration.txt (hunk 1 above)
Comment (pszkotak) [reply openthread/lib/cortex-m4/soft-float/v1.1/ftd/cc310/openthread_lib_configuration.txt:49]: Done

**Review by edmont — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by shantha-14 — APPROVED** [review]

**Pull request merged**